plugins/discovery-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeService.java

```diff
@@ -39,9 +44,18 @@
     }
 
     static public final class Discovery {
-        public static final String REFRESH = "discovery.azure.refresh_interval";
+        public static final Setting<TimeValue> REFRESH_SETTING = Setting.positiveTimeSetting("discovery.azure.refresh_interval", TimeValue.timeValueSeconds(0), false, Setting.Scope.CLUSTER);
+
+        public static final Setting<AzureUnicastHostsProvider.HostType> HOST_TYPE_SETTING = new Setting<>(
+            "discovery.azure.host.type", AzureUnicastHostsProvider.HostType.PRIVATE_IP.name(),
+            strHostType -> {
+                AzureUnicastHostsProvider.HostType hostType = AzureUnicastHostsProvider.HostType.fromString(strHostType.toUpperCase(Locale.ROOT));
```

Comment:
I know you don't have written the code but I think it would be even cleaner if `AzureUnicastHostsProvider.HostType.fromString(String)` would already throw an `IllegalArgumentException` for invalid strings instead of returning `null`. wdyt?